app/controllers/files/download_controller.rb

```diff
@@ -1,6 +1,15 @@
 class Files::DownloadController < Files::FilesController
   skip_before_filter :set_identity
-  before_filter :set_file, only: :download_url
+  before_filter :set_file, except: :download
+
+  def preview
+    if @file.identity.user == current_user
+      # send the file to the user
+      send_data @file.identity.get_file(@file.foreign_ref), disposition: 'inline',
```

Comment:
Would it be possible to cache the files in the `tmp` folder somehow? Going back and forth between a file feels slow now. If we were to cache the files, we could perhaps also use http response headers (`304 Not Modified`) to speed up the system.